services/src/main/java/org/keycloak/protocol/oid4vc/OID4VCLoginProtocolFactory.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Override
+    public void postInit(KeycloakSessionFactory factory) {
+        // no-op
+    }
+
+    @Override
+    public void close() {
+        // no-op
+    }
+
+    @Override
+    public Map<String, ProtocolMapperModel> getBuiltinMappers() {
+        return builtins;
+    }
+
+    private void addServiceFromComponent(Map<Format, VerifiableCredentialsSigningService> signingServices, KeycloakSession keycloakSession, ComponentModel componentModel) {
+        ProviderFactory<VerifiableCredentialsSigningService> factory = keycloakSession
+                .getKeycloakSessionFactory()
+                .getProviderFactory(VerifiableCredentialsSigningService.class, componentModel.getProviderId());
+        if (factory instanceof VCSigningServiceProviderFactory sspf) {
+            signingServices.put(sspf.supportedFormat(), sspf.create(keycloakSession, componentModel));
+        } else {
+            throw new IllegalArgumentException(String.format("The component %s is not a VerifiableCredentialsSigningServiceProviderFactory", componentModel.getProviderId()));
+        }
+
+    }
+
+    @Override
+    public Object createProtocolEndpoint(KeycloakSession keycloakSession, EventBuilder event) {
+
+        Map<Format, VerifiableCredentialsSigningService> signingServices = new EnumMap<>(Format.class);
+        RealmModel realm = keycloakSession.getContext().getRealm();
+        realm.getComponentsStream(realm.getId(), VerifiableCredentialsSigningService.class.getName())
+                .forEach(cm -> addServiceFromComponent(signingServices, keycloakSession, cm));
+
+        RealmModel realmModel = keycloakSession.getContext().getRealm();
+        String issuerDid = Optional.ofNullable(realmModel.getAttribute(ISSUER_DID_REALM_ATTRIBUTE_KEY))
+                .orElseThrow(() -> new VCIssuerException("No issuer-did  configured."));
+        int preAuthorizedCodeLifespan = Optional.ofNullable(realmModel.getAttribute(CODE_LIFESPAN_REALM_ATTRIBUTE_KEY))
+                .map(Integer::valueOf)
+                .orElse(DEFAULT_CODE_LIFESPAN_S);
+
+        return new OID4VCIssuerEndpoint(
+                keycloakSession,
+                issuerDid,
+                signingServices,
+                new AppAuthManager.BearerTokenAuthenticator(keycloakSession),
+                OBJECT_MAPPER,
+                new OffsetTimeProvider(),
+                preAuthorizedCodeLifespan);
+    }
+
+    @Override
+    public void createDefaultClientScopes(RealmModel newRealm, boolean addScopesToExistingClients) {
+        LOGGER.debugf("Create default scopes for realm %s", newRealm.getName());
+
+        ClientScopeModel naturalPersonScope = KeycloakModelUtils.getClientScopeByName(newRealm, "natural_person");
+        if (naturalPersonScope == null) {
+            LOGGER.debug("Add natural person scope");
+            naturalPersonScope = newRealm.addClientScope("natural_person");
```

Comment:
I am thinking if it makes sense to name this client scope as `oid4vc_natural_person` to make it more clear that it is about `oid4vc` ? But not strong opinion on this as protocol is visible anyway in the table of client scopes. @tnorimat WDYT?